### PR TITLE
refactor(swap): handle API changes, fix lrange

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

File: packages/shared/src/node-apis/redis.ts [hunk 1]
```diff
@@ -46,13 +43,15 @@ type PolkadotBroadcast = ChainBroadcast<'Polkadot'>;
...
-  value: number.transform((value) =>
-    new BigNumber(value).shiftedBy(assetDecimals.BTC).toString(),
-  ),
-  tx_hash: string.transform((value) => `0x${value}` as const),
-});
+const mempoolTransaction = jsonString.pipe(
+  z.object({
+    confirmations: number,
+    value: number.transform((value) =>
+      new BigNumber(value).shiftedBy(assetDecimals.BTC).toString(),
```
Comment (niklasnatter) [inline packages/shared/src/node-apis/redis.ts:50]: i think you changed this, do we still need the shift?

File: packages/shared/src/node-apis/redis.ts [hunk 2]
```diff
@@ -89,19 +88,17 @@ export default class RedisClient {
...
-      ? deposits
-          .parse(JSON.parse(value))
-          .filter((deposit) => deposit.asset.asset === asset)
-          .sort(sortDepositAscending)
-      : [];
+    const deposits = await this.client.lrange(key, 0, -1);
+    return deposits
+      .map((deposit) => depositSchema.parse(deposit))
+      .filter((deposit) => deposit.asset === asset)
+      .sort(sortDepositAscending);
```
Comment (niklasnatter) [inline packages/shared/src/node-apis/redis.ts:95]: as this is a linked list and we are using `rpush` to store it, i think we would not need to sort

Comment (acdibble) [reply packages/shared/src/node-apis/redis.ts:92]: yeah we probably never needed to but we're not guaranteed to always be using `rpush` to store it

File: packages/shared/src/node-apis/redis.ts (hunk 1 above)
Comment (acdibble) [reply packages/shared/src/node-apis/redis.ts]: fixed

**Review by jerryafr — APPROVED** [review]

**Review by niklasnatter — APPROVED** [review]
cool!

**Pull request merged**